src/server.py

```diff
@@ -1,8 +1,214 @@
+    
+    # Format the classes properly
+    classes_formatted = []
+    for class_detail in classes["groups"]:
+        cur_class_details = {
+            "type": class_detail["type"],
+            "classes": []
+        }
+        
+        classes = []
+        for _class in class_detail["classes"]:
+            cur_class = {
+                "number": _class["class_nbr"],
+                "section": _class["section"],
+                "capacity": { # Not in the spec, but useful to have
+                    "size": _class["size"],
+                    "enrolled": _class["enrolled"],
+                    "available": _class["available"] if _class["available"] != "FULL" else 0,
+                },
+                "notes": _class["notes"] if "notes" in _class.keys() else [], # Leaving this as it is in the API
+                "meetings": []
+            }
+            
+            meetings = []
+            for meeting in _class["meetings"]:
+                meetings.append({
+                    "day": meeting["days"],
+                    "date": meeting_date_convert(meeting["dates"]),
+                    "time": {
+                        "start": meeting_time_convert(meeting["start_time"]),
+                        "end": meeting_time_convert(meeting["end_time"])
+                    },
+                    "location": meeting["location"]
+                })
+            
+            cur_class["meetings"] = meetings
+            classes.append(cur_class)
+            
+        cur_class_details["classes"] = classes
+        classes_formatted.append(cur_class_details)
+    
+    course_info = {
+        "id": id,
+        "name": {
+            "subject": details["SUBJECT"],
+            "code": details["CATALOG_NBR"],
+            "title": details["COURSE_TITLE"]
+            # Below removed because such info is not in the courseplanner api
+            #"other_names": ["ADDS", "DSA", "Data structures and algorithms"], # case insensitive in frontend search
+        },
+        "class_number": details["CLASS_NBR"],
+        "year": details["YEAR"],
+        "term": details["TERM_DESCR"], # Could use the "TERM" field but would then need to match the given term code
+        "campus": details["CAMPUS"],
+        "career": details["ACAD_CAREER_DESCR"], # Using a full text version instead of a code (eg UGRD)
+        "units": details["UNITS"],
+        "requirement": {
+            "restriction": details["RESTRICTION_TXT"] if details["RESTRICTION"] == "Y" else None,
+            # Requisites are parsed into a dict with the original string and parsed subjects
+            "prerequisite": parse_requisites(details["PRE_REQUISITE"]),
+            "corequisite": parse_requisites(details["CO_REQUISITE"]),
+            "assumed_knowledge": parse_requisites(details["ASSUMED_KNOWLEDGE"]),
+            "incompatible": parse_requisites(details["INCOMPATIBLE"])
+        },
+        # Not in the spec
```

Comment:
As in the spec outlined in #3?

Reply:
I was referring to the below lines (description, critical_dates, etc) as they weren't outlined in the spec, however I added them as I thought they could be useful info for our API to have